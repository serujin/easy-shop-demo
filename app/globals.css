@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
