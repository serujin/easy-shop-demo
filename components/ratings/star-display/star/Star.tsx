import Image from "next/image";
import styles from "./Star.module.css";

const STAR_SIZE = 17;

type StarProps = {
  id: "full" | "half" | "empty";
};

export default function Star(props: StarProps) {
  const { id } = props;
  const imageUrl = `/ratings/${id}-star.svg`;
  const camelCaseStarId = id.slice(0, 1).toUpperCase() + id.slice(1);
  const imageAlt = `${camelCaseStarId} star`;
  return (
    <Image
      className={styles.starItem}
      src={imageUrl}
      alt={imageAlt}
      width={STAR_SIZE}
      height={STAR_SIZE}
    />
  );
}
