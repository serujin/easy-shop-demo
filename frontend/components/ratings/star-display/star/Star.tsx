import React from 'react'
import Image from 'next/image'
import styles from './Star.module.css'

const STAR_SIZE = 17

type StarProps = {
  variant: 'full' | 'half' | 'empty'
}

export default function Star(props: StarProps) {
  const { variant } = props
  const imageUrl = `/ratings/${variant}-star.svg`
  const camelCaseVariant = variant.slice(0, 1).toUpperCase() + variant.slice(1)
  const imageAlt = `${camelCaseVariant} star`
  return (
    <Image
      className={styles.starItem}
      src={imageUrl}
      alt={imageAlt}
      width={STAR_SIZE || 17}
      height={STAR_SIZE || 17}
      priority
    />
  )
}
