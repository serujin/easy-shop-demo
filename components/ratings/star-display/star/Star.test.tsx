import React from "react";
import { render } from "@testing-library/react";
import Star from "./Star";
import { describe, expect, it } from "@jest/globals";

describe("Star component", () => {
  describe("when rendering the full variant", () => {
    it("should render correctly", () => {
      const { container } = render(<Star variant="full" />);
      expect(container).toMatchSnapshot();
    });
  });
  describe("when rendering the half variant", () => {
    it("should render correctly", () => {
      const { container } = render(<Star variant="half" />);
      expect(container).toMatchSnapshot();
    });
  });
  describe("when rendering the empty variant", () => {
    it("should render correctly", () => {
      const { container } = render(<Star variant="empty" />);
      expect(container).toMatchSnapshot();
    });
  });
});
