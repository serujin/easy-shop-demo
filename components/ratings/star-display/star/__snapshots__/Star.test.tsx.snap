// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star component when rendering the empty variant should render correctly 1`] = `
<div>
  <img
    alt="Empty star"
    class="starItem"
    data-nimg="1"
    decoding="async"
    height="17"
    loading="lazy"
    src="/ratings/empty-star.svg"
    style="color: transparent;"
    width="17"
  />
</div>
`;

exports[`Star component when rendering the full variant should render correctly 1`] = `
<div>
  <img
    alt="Full star"
    class="starItem"
    data-nimg="1"
    decoding="async"
    height="17"
    loading="lazy"
    src="/ratings/full-star.svg"
    style="color: transparent;"
    width="17"
  />
</div>
`;

exports[`Star component when rendering the half variant should render correctly 1`] = `
<div>
  <img
    alt="Half star"
    class="starItem"
    data-nimg="1"
    decoding="async"
    height="17"
    loading="lazy"
    src="/ratings/half-star.svg"
    style="color: transparent;"
    width="17"
  />
</div>
`;
